### ISSUE TITLE
Modularized Morphing of Neural Networks

### ISSUE THREAD

**Issue by sile** [sile/readings#110]
URL
-----

https://arxiv.org/abs/1701.03281

Abstract
----------

> In this work we study the problem of network morphism, an effective learning scheme to morph a well-trained neural network to a new one with the network function completely preserved. Different from existing work where basic morphing types on the layer level were addressed, we target at the central problem of network morphism at a higher level, i.e., how a convolutional layer can be morphed into an arbitrary module of a neural network. To simplify the representation of a network, we abstract a module as a graph with blobs as vertices and convolutional layers as edges, based on which the morphing process is able to be formulated as a graph transformation problem. Two atomic morphing operations are introduced to compose the graphs, based on which modules are classified into two families, i.e., simple morphable modules and complex modules. We present practical morphing solutions for both of these two families, and prove that any reasonable module can be morphed from a single convolutional layer. Extensive experiments have been conducted based on the state-of-the-art ResNet on benchmark datasets, and the effectiveness of the proposed solution has been verified.

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
- (3) Network Morphism via Graph Abstraction
  - (3-1) Background and Basic Notations
  - (3-2) Atomic Network Morphism
  - (3-3) Graph Abstraction
  - (3-4) The Compatibility of Network MOrphism Equation
  - (3-5) Simple Morphable Module
  - (3-6) Modular Network Morphism Theorem
  - (3-7) Non-linearity and Batch Normalization in Modular Network Morphism
- (4) Experimental Results
  - (4-1) Network Architectures of Modular Network Morphism
  - (4-2) Experimental Results on the CIFAR10 Dataset
  - (4-3) Experimental Results on the CIFAR100 Dataset
  - (4-4) Experimental Results on the ImageNet Dataset
- (5) Conclusions

BibTeX
------

```bibtex
@misc{wei2017modularized,
    title={Modularized Morphing of Neural Networks},
    author={Tao Wei and Changhu Wang and Chang Wen Chen},
    year={2017},
    eprint={1701.03281},
    archivePrefix={arXiv},
    primaryClass={cs.LG}
}
```

See Also
----------

- GitHub:
- #107
